### PR TITLE
Update Readme with pre-commit

### PR DESCRIPTION
**Pull request by SeanMooney**
- This change adds more detail on how to configure pre-commit
- This change adds a pre-commit badge to show branch status.

### CODE REVIEW

**Comment by bogdando** [comment]
I have this failure (it could be my env specific problem though)
```
make-generate............................................................Failed
- hook id: make-generate
- exit code: 2

test -s /opt/go/src/github.com/openstack-k8s-operators/nova-operator/bin/controller-gen || GOBIN=/opt/go/src/github.com/openstack-k8s-operators/nova-operator/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.9.2
/opt/go/src/github.com/openstack-k8s-operators/nova-operator/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
Error: err: exit status 1: stderr: go: -mod may only be set to readonly when in workspace mode, but it is set to "mod"
        Remove the -mod flag to use the default readonly value,
        or set GOWORK=off to disable workspace mode.
```

File: README.md [hunk 1]
```diff
@@ -1,18 +1,102 @@
...
+```sh
+sudo dnf install -y golang
+```
+
+#### installing in a virutal env
+
+```sh
+python3 -m venv .venv
+. .venv/bin/activate
+python3 -m TODO(user): An in-depth paragraph about your project and overview of useTODO(user): An in-depth paragraph about your project and overview of useTODO(user): An in-depth paragraph about your project and overview of usepip install pre-commit
```
Comment (gibizer) [inline README.md:38]: the `python3 -m` and a bunch of existing `TODO`s are now intermingled here.

File: README.md [hunk 2]
```diff
@@ -1,18 +1,102 @@
...
+add "$(go env GOPATH)/bin" to your path via ~/.bashrc
+
+```sh
+if ! [[ "$PATH" =~ "$(go env GOPATH)/bin" ]]
+then
+    export PATH=${PATH}:$(go env GOPATH)/bin
+fi
+```
+
+confrim golangci-lint is installed
```
Comment (gibizer) [inline README.md:59]: confirm

**Review by gibizer — CHANGES_REQUESTED** [review]

**Comment by gibizer** [comment]
> I have this failure (it could be my env specific problem though)
> 
> ```
> make-generate............................................................Failed
> - hook id: make-generate
> - exit code: 2
> 
> test -s /opt/go/src/github.com/openstack-k8s-operators/nova-operator/bin/controller-gen || GOBIN=/opt/go/src/github.com/openstack-k8s-operators/nova-operator/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.9.2
> /opt/go/src/github.com/openstack-k8s-operators/nova-operator/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
> Error: err: exit status 1: stderr: go: -mod may only be set to readonly when in workspace mode, but it is set to "mod"
>         Remove the -mod flag to use the default readonly value,
>         or set GOWORK=off to disable workspace mode.
> ```

what is your go version?

> $ go version
> go version go1.18.5 linux/amd64
>

File: README.md (hunk 1 above)
Comment (SeanMooney) [reply README.md:38]: oh odd i wonder how that happened ill fix that

**Review by gibizer — APPROVED** [review]
/lgtm
We can troubleshoot @bogdando 's env issue separately.

**Comment by gibizer** [comment]
> On my side, the problem was in `GOFLAGS=-mod=mod`. If I unset it, everything works

Thanks for stating the fix. I checked I have no GOFLAGS in my env.

**Pull request merged**